### DISCUSSION TITLE
Can anyone please help why am getting the error -> javax.net.ssl.SSLHandshakeException: Received fatal alert: protocol_version?

### DISCUSSION THREAD

**Discussion by mfarooquijlg** [Azure/azure-iot-sdk-java#1781] [General]
Hi

I have been debugging this issue since over days and tried many different options but still no luck. What exactly is the root cause of the error here? I am using this following example to provision the device and getting the below error 

https://github.com/Azure/azure-iot-sdk-java/blob/main/provisioning/provisioning-device-client-samples/provisioning-X509-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/ProvisioningX509Sample.java

ERROR STACK TRACE
java.util.concurrent.ExecutionException: com.microsoft.azure.sdk.iot.provisioning.device.internal.exceptions.ProvisioningDeviceTransportException: javax.net.ssl.SSLHandshakeException: Received fatal alert: protocol_version

**Comment by timtay-microsoft** [comment]
Do you know what TLS version your device is configured to use? Based on that error message, my guess is that your device uses an older TLS version that Device Provisioning Service does not support. For context, Device Provisioning Service only currently supports TLS version 1.2 and the SDK will attempt to use TLS1.2 by default
